app/views/shared/_error_messages.html.erb

```diff
@@ -0,0 +1,13 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      <%= t ".intro" %>
+      <%= pluralize(@user.errors.count, t(".error")) %>.
```

Comment:
Phân biệt lenght, count, size giúp anh

Reply:
+ count đếm số bản ghi trong database của bảng, tương tự truy vấn SELECT COUNT(*).
User.count
=> 40
+ length sẽ load toàn bộ giá trị trong database vào 1 mảng và đo kích thước của mảng đó
user = User.all
user.length
=> 40
User.length
undefined method `length'
+ size sẽ đếm số bản ghi trong database của bảng bằng truy vấn SELECT COUNT(*) khi các bản ghi chưa được lưu vào mảng
User.all.size
=> 40
hoặc đếm số bản ghi khi các bản ghi đã được lưu trong mảng
user = User.all
user.size
=> 40
size không thể dùng như count
User.size
undefined method `size'